src/test/ui/parser/lex-bad-char-literals-3.stderr

```diff
@@ -18,5 +18,15 @@ help: if you meant to write a `str` literal, use double quotes
 LL |     let ch: &str = "●●";
    |                    ^^^^
 
-error: aborting due to 2 previous errors
+error[E0308]: mismatched types
+  --> $DIR/lex-bad-char-literals-3.rs:5:20
+   |
+LL |     let ch: &str = '●●';
+   |                    ^^^^ expected &str, found char
```

Comment:
Literals with errors need to be converted to `Lit::Err` to reduce further errors, doing that only during parsing would certainly be good enough.

In general adding `Lit::Err` at token level probably wasn't the best idea, it's only necessary in AST.
So perhaps it can be removed (later, or now if it's easy after this refactoring).

Reply:
If it doesn't hurt pattern-matching too much, `Result<Lit, ErrorReported>` would probably work.